flo-runner/src/test/java/com/spotify/flo/context/FloRunnerTest.java

```diff
@@ -371,6 +387,137 @@ public void tasksRunInProcesses() throws Exception {
     assertThat(uniqueJvms, not(contains(mainJvm)));
   }
 
+  @Test
+  public void isTestShouldBeTrueInTestScope() throws Exception {
+    try (TestScope ts = FloTesting.scope()) {
+      final Task<Boolean> isTest = Task.named("task").ofType(Boolean.class)
+          .process(FloTesting::isTest);
```

Comment:
How does this work?

I might just not understand the grpc Context class enough. It seems like it only has a `ThreadLocalContextStorage` which leads me to think that whatever `TestScope` (with accompanying `TestContext`) would be bound to the test runner thread and not the worker threads.

Does this work because of the propagation in `TracingContext`? If so, maybe there should be an explicit link in the javadoc mentioning the dependency.

Reply:
This works because `AsyncContext` [propagates](https://github.com/spotify/flo/blob/master/flo-workflow/src/main/java/com/spotify/flo/context/AsyncContext.java#L47) the grpc `Context` to worker threads.

Reply:
Aha, that't pretty neat!